stablehlo/dialect/AssemblyFormat.cpp

```diff
@@ -280,6 +280,30 @@ ParseResult parseSelectOpType(OpAsmParser& parser, Type& pred, Type& onTrue,
 // Attribute Printers and Parsers
 //===----------------------------------------------------------------------===//
 
+void printDenseI64Array(OpAsmPrinter& p, Operation* op,
+                        DenseIntElementsAttr attr) {
+  assert(attr.getType().getRank() == 1);  // Only works for 1D array attrs.
```

Comment:
Given that asserts can be elided depending on compiler configuration, let's use report_fatal_error instead.

Reply:
Updated to use `report_fatal_error`. Tested by incorrectly modifying collective_permute to use this custom printer, and it triggered the error `LLVM ERROR: printDenseI64Array only supports rank-1 arrays`.